test/unit/subcommands/test_stop_subcommand.py

```diff
@@ -1,167 +1,175 @@
+    def test_stop_subcommand_doesnt_kill_pid_if_pid_file_and_pid_exist_but_command_isnt_whitelisted(self):
+        # Arrange
+        self._mock_os_path_exists.return_value = True
+        self._mock_psutil_pid_exists.return_value = True
+        master_process = Mock(psutil.Process)
+        master_process.cmdline.return_value = ['python', './foo.py']
+        slave_process = Mock(psutil.Process)
+        slave_process.cmdline.return_value = ['python', './bar.py']
+        self._mock_psutil_process.side_effect = [
+            slave_process,
+            master_process,
+        ]
+
+        # Act
+        self._stop_subcommand.run(None)
+
+        # Assert
+        self.assertFalse(self._mock_os_kill.called)
+
+    def _create_mock_process(self, pid, child_processes=None, cmdline=None):
+        proc = Mock(psutil.Process)
+        proc.pid = pid
+        proc.is_running.return_value = True
         if cmdline:
-            m.cmdline.return_value = cmdline
-
-        m.children.return_value = children if children else []
-        return m
-
-    def test_stop_subcommand_kills_pid_with_sigterm_if_pid_file_and_pid_exist_and_command_is_whitelisted(self):
-        main_proc_pid = 9999
-        threshold = NUM_OF_IS_RUNNING_CHECKS_TILL_SIGKILL - 1
-        child_pid = main_proc_pid + 1
-
-        self.setup_environment_for_os_kill(main_proc_pid)
-        self.setup_main_and_child_proc(main_proc_pid, threshold, child_pid, threshold)
-
-        stop_subcommand = StopSubcommand()
-        stop_subcommand._kill_pid_in_file_if_exists('/tmp/pid_file_path.pid')
-
-        self.os_kill_patch.assert_called_with(main_proc_pid, signal.SIGTERM)
-        self.assertEquals(2, self.os_kill_patch.call_count)
-
-    def test_stop_subcommand_kills_main_proc_with_sigkill_if_still_running_after_sigterm(self):
-        main_proc_pid = 9999
-        child_proc_pid = main_proc_pid + 1
-        main_threshold = NUM_OF_IS_RUNNING_CHECKS_TILL_SIGKILL
-        child_threshold = NUM_OF_IS_RUNNING_CHECKS_TILL_SIGKILL - 1
-
-        self.setup_environment_for_os_kill(main_proc_pid)
-        self.setup_main_and_child_proc(main_proc_pid,main_threshold, child_proc_pid, child_threshold)
-
-        stop_subcommand = StopSubcommand()
-        stop_subcommand._kill_pid_in_file_if_exists('/tmp/pid_file_path.pid')
-
-        self.os_kill_patch.assert_called_with(main_proc_pid, signal.SIGKILL)
-        self.assertEquals(3, self.os_kill_patch.call_count)
+            proc.cmdline.return_value = cmdline
+        proc.children.return_value = child_processes if child_processes else []
+        return proc
 
-    def test_stop_subcommand_kills_child_proc_with_sigkill_if_still_running_after_sigterm(self):
-        main_proc_pid = 9999
-        main_threshold = NUM_OF_IS_RUNNING_CHECKS_TILL_SIGKILL - 1
-        child_threshold = NUM_OF_IS_RUNNING_CHECKS_TILL_SIGKILL
-        child_pid = main_proc_pid + 1
+    def _expect_kill_processes(self):
```

Comment:
What do you think of renaming this to something other than "expect". I was thinking that it's not really setting up any expectations or making any assertions.